src/plugins/vis_types/pie/public/pie_component.tsx

```diff
@@ -186,10 +187,10 @@ const PieComponent = (props: PieComponentProps) => {
   const { visData, visParams, services, syncColors } = props;
 
   function getSliceValue(d: Datum, metricColumn: DatatableColumn) {
-    if (typeof d[metricColumn.id] === 'number' && d[metricColumn.id] !== 0) {
+    if (typeof d[metricColumn.id] === 'number') {
       return d[metricColumn.id];
     }
-    return Number.EPSILON;
+    return 0;
```

Comment:
Nit, it could be a single expression without tedious control structures, while maybe also extracting the common constant (for code reading; otherwise a JS engine likely hoists common constant expressions anyway):
```suggestion
const value = d[metricColumn.id];
return Number.isFinite(value) && value >= 0 ? value : 0;
```

Reply:
I added the positive value check, though charts will handle negative values by throwing them out anyway (so it's optional, but helps clarify for Kibana code readers what to expect)

Reply:
I agree, I like it :)